include/gbinder_types.h

```diff
@@ -167,16 +167,66 @@ GBinderLocalReply*
 
 #define GBINDER_TX_FLAG_ONEWAY (0x01)
 
+/* other possible errors from system/core/libutils/include/utils/Errors.h */
 typedef enum gbinder_status {
     GBINDER_STATUS_OK = 0,
     GBINDER_STATUS_FAILED,
-    GBINDER_STATUS_DEAD_OBJECT
+    GBINDER_STATUS_UNKNOWN_ERROR       = (-2147483647-1), // INT32_MIN value
```

Comment:
Yes that makes sense, though it's more the other way around - we need to be able to return these codes like UNKNOWN_TRANSACTION and PERMISSION_DENIED in order to accurately flag errors in calls to binder services. Currently returning GBINDER_STATUS_FAILED in a GBinderLocalReply causes 'Unknown binder error code. 0x1' when interpreted [here](https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/jni/android_util_Binder.cpp;drc=c4739e7ff8a2049dd5d783cca40e076d7edb6c92;l=1018). So they're not always being caught right now.

Reply:
If that's the value passed to `gbinder_driver_reply_status()` then what about replacing `GBINDER_STATUS_FAILED` with `INT32_MIN` and `GBINDER_STATUS_DEAD_OBJECT` with `-EPIPE`. Would that do it?

Reply:
Nearly. We should be sure that the actual values will be completely internal, i.e. they're translated back when returned by Android services too.

Can we not add the others to the end of the enum without breaking anything though? We really need GBINDER_STATUS_PERMISSION_DENIED (-EPERM) to get these tests working nicely, but I can see GBINDER_STATUS_UNKNOWN_TRANSACTION (-EBADMSG) should be what everything returns in the default case of all transaction code switches. Really any of them can be returned by Android services.

Reply:
If these are going to be completely internal, then they would have to be replaced in requests too. I guess gbinder_driver_txstatus would be the place to do that?

Reply:
Yes, a few additional public enum values feels like the way to go and `gbinder_driver_txstatus()` does seem to be the right place for the conversion, although I'm a bit puzzled by the switch under the "Filter out special cases" comment - it seems to do the conversion wrong way, e.g. replacing `GBINDER_STATUS_DEAD_OBJECT` with `-EFAULT`. Now that I'm looking at it, I'd say it should convert e.g. `-EPIPE` to `GBINDER_STATUS_DEAD_OBJECT` since that's what's going to be returned as a status by e.g. `gbinder_client_transact()` and friends (unless I'm missing something)

Reply:
Ah I'm glad I wasn't the only one confused by that. I had no idea where that EFAULT had come from. So my plan is to convert FAILED and DEAD_OBJECT in there and reply_status, and add the others with their actual values so that's not necessary. Is that ok, or would you rather they were all converted?